src/app/static/js/editor/postscreenshot.js

```diff
@@ -30,19 +27,56 @@ async function getFileData(imgUrl) {
 }
 
 async function postToGithub(data) {
-  let xhr = new XMLHttpRequest();
-  let filename =
-    Math.random()
-      .toString(36)
-      .substring(3) + ".png";
-
-  localStorage.setItem("filename", filename);
-  apiurl =
-    "https://api.github.com/repos/" +
-    repositoryNameWithOwner +
-    "/contents/" +
-    filename;
-  xhr.open("PUT", apiurl);
-  xhr.setRequestHeader("Authorization", "Token " + githubToken);
-  xhr.send(JSON.stringify(data));
+  let fileurl = null;
+  $.ajax({
+        type: "POST",
+        enctype: 'multipart/form-data',
+        url: "/app/post_to_github/",
+        dataType: 'json',
+        timeout: 600000,
+        data: data,
+        success: function (data) {
+            var githubCode = data.statusCode;
+            if(githubCode == '201'){
+              fileurl = data.fileurl
```

Comment:
If this is an asynchronous call, the function will return null for fileurl if it returns before the ajax call is returned.

A simple fix would be to add  `async: false`